### PR TITLE
Gem sim validation porting 76 x only occupy plots

### PR DESCRIPTION
**Pull request by geonmo**
### Hit Validation module for GEM Detector after Run2
1. This PR include Validation/MuonGEMHits, MuonGEMDigis, MuonGEMRecHits module to validate GEM hit simulation.
   - MuonGEMHits module is related to SimHit on GEM Detector(Run2019/2023).
   - MuonGEMDigis module is related to Digi step about GEM Strip, pad and Copad.
   - MuonGEMRechits module is related to local reco hit on GEM Detector. 
#### This PR does not affect to running other modules.
- These module are only running with specific customize options.

Remake PR due to wrong push to my branch and too old.
It is same PR #10723 except some fix typo or remove commented lines.

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by geonmo** [comment]
@deguio Hello, Federico. Could you approve this PR? I will make another PR for efficiency plots.

File: Validation/MuonGEMDigis/src/GEMCheckGeometry.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        temp_me->setBinLabel(4,"St1,La2_odd",2);
+        temp_me->setBinLabel(5,"St2,La1_even",2);
+        temp_me->setBinLabel(6,"St2,La1_odd",2);
+        temp_me->setBinLabel(7,"St2,La2_even",2);
+        temp_me->setBinLabel(8,"St2,La2_odd",2);
+        temp_me->setBinLabel(9,"St3,La1_even",2);
+        temp_me->setBinLabel(10,"St3,La1_odd",2);
+        temp_me->setBinLabel(11,"St3,La2_even",2);
+        temp_me->setBinLabel(12,"St3,La2_odd",2);
+        theStdPlots.insert( std::map< UInt_t, MonitorElement*>::value_type(name.Hash(), temp_me ));
```
Comment (rovere) [inline Validation/MuonGEMDigis/src/GEMCheckGeometry.cc]: I do not see the point of using an ordinary map as a hash-table?? What is the meaning of having the hash-keys ordered (and the time spent to order them)? You get the worst of the 2 from this approach, IMHO. Use directly an unordered_map.

**Comment by deguio** [comment]
ciao @geonmo 
I am in the process of reviewing this PR finally..  (sorry for being so late)
a couple of general comments in addition to marco's comment:

- there are few cases where memory is allocated with new but never deleted. see for example:
https://github.com/cms-sw/cmssw/pull/11123/files#diff-0b49d714d6e00574dcfa3e3e57202922R63
- I am not sure which criterion you used to name cfi vs cff files
- the number of histograms (bins) booked is reasonable. I hope that the detailed plots will be considered for local running only. see for example: https://github.com/cms-sw/cmssw/pull/11123/files#diff-b0e32bae891ed958cf3bf1132d1b22c6R33  (4.320.000 bins!)
- is this class under development? huge portion of code is commented out. https://github.com/cms-sw/cmssw/pull/11123/files#diff-5e6117ecb55ff174214e1cabd9f2cb3fR97


This PR adds code which is not included in any central sequence, but has been tested locally. more details at:
https://indico.cern.ch/event/446270/contribution/2/attachments/1154267/1658925/Presentation_on_DQM_MeetingGEM_Validation_modules.pdf

File: Validation/MuonGEMDigis/src/GEMCheckGeometry.cc (hunk 1 above)
Comment (geonmo) [inline Validation/MuonGEMDigis/src/GEMCheckGeometry.cc]: Hello, rovere. I used "hash" function to get unique value from "string". Frankly speaking, I have a some trouble due to "const char*" comparision problem. So, I used Hash function to solve temporary. In addition, I heared unorderd_map is slower than map for small pairs. I will fix to use "map" with "std::string". Thanks for interest.

Comment (rovere) [inline Validation/MuonGEMDigis/src/GEMCheckGeometry.cc]: No, using std:string as key in the map is even worse. Either leave it this way, or use an std::unordered_map with a string as key: string should natively have hashing specialisation available to you. And std hash tables handle collisions for you.

**Comment by geonmo** [comment]
@deguio Hello, Federico.  
- For Comment #1, 
 - Ok , I will fix it as soon as possible. However, it is a big holyday season in Korea. So, I can update next weeks or later.
- For Comment #2, 
 - I used cfi files for dummy values(to show all variables) and cff files for real values(to hide some variables). However, they were written discursively. I will cleaning and refactoring to proper standard.
- For Comment #3,
 - detailPlots are only used for gem software developer. It never run on central process.(If so, it is a bug.)
- For Comment #4, 
 - That file is related to efficiency plots(actullay, it is used to draw detail efficiency plots.). I commented out for this PR. However, I will uncomment and fix when I make a PR for efficiency plots.

Many thanks for interest and review.
I will update as soon as possible as your comment. 
Thanks,

**Comment by geonmo** [comment]
Hello, @deguio 
I checked about to do not delete memory allocation.
I made some histogram and profile using "new" keyword and all cases will be booked on IBooker.
As I know, ibooker's book functions need histogram's pointers and ibooker will deleted if no more histogram is required, isn't it?

**Comment by geonmo** [comment]
Hello, @deguio and @rovere .
I took a mistake to missing rebase before submitting a PR.
I will re-make new PR with same contents.

**Pull request closed without merging**